src/ScriptedInputSource.cpp

```diff
@@ -84,6 +84,40 @@ PlayerInputAbs ScriptedInputSource::getNextInput()
 		state.swapSides();
 	}
 
+	// if the ball is on the bots side, decrease the error timers more quickly.
+	// if we did not do this, then the difficulty would have almost no effect on
+	// any plays made by the player from the back of their half, as the counters
+	// would have reached zero by the time the ball arrives at the bot's half. This
+	// way, we can give the counters larger initial values, and still don't get completely
+	// stupid play.
+	if(state.getBallPosition().x < 400) {
+		--mBallPosErrorTimer;
+		--mReactionTime;
+	}
+
+	// decrement counters regularly
+	--mReactionTime;
+	--mBallPosErrorTimer;
```

Comment:
This timers can be negative. When simulating good bots (for a long time) this may result in a integer overflow.

Reply:
true, if the game runs for more than 7000h without resetting these times (i.e. with no one scoring), this would overflow.